c++/include/orc/Type.hh

```diff
@@ -65,6 +65,7 @@ namespace orc {
     virtual std::vector<std::string> getAttributeKeys() const = 0;
     virtual std::string getAttributeValue(const std::string& key) const = 0;
     virtual std::string toString() const = 0;
+    virtual const Type* getTypeByColumnId(uint64_t colId) const = 0;
```

Comment:
We have supported C++17 now. It can be replaced by std::optional. Or we need to add a comment to say what will happen if the column does not exist.